self-core-impl/src/main/java/com/selfxdsd/core/GithubStars.java

```diff
@@ -78,12 +81,99 @@ public boolean add() {
         } else {
             starred = false;
             LOG.error(
-                "Unexpected status when starring repo " + repoName
-                + " of user " + owner + " ]"
-                + "Expected 204 NO CONTENT, but got "
-                + response.statusCode()
+                "Unexpected status when starring repo "
+                    + this.holder.getRepo()
+                    + " of user " + this.holder.getOwner() + " ]"
+                    + "Expected 204 NO CONTENT, but got "
+                    + response.statusCode()
             );
         }
         return starred;
     }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * See doc <a href="https://docs.github.com/en/rest/reference/activity#check-if-a-repository-is-starred-by-the-authenticated-user">here</a>.
+     */
+    @Override
+    public boolean isStarred() {
+        LOG.debug(
+            "Check if Github repository " + this.holder.getRepo()
+                + " is starred by current authenticated user."
+        );
+
+        final boolean isStarred;
+        final Resource response = this.resources.get(
+            URI.create(this.starsUri.toString()),
+            () -> Map.of("Accept", List.of("application/vnd.github.v3+json"))
+        );
+        if (response.statusCode() == HttpURLConnection.HTTP_NO_CONTENT) {
+            isStarred = true;
+            LOG.debug("Repo is starred by current authenticated user.");
+        } else {
+            isStarred = false;
+            if (response.statusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+                LOG.debug("Repo is not starred by current authenticated user.");
+            } else {
+                LOG.error(
+                    "Unexpected status when checking if repo "
+                        + this.holder.getRepo()
+                        + " of user " + this.holder.getOwner()
+                        + " is starred by current authenticated user."
+                        + " Expected 204 NO CONTENT or 404 HTTP_NOT_FOUND,"
+                        + " but got " + response.statusCode()
+                );
+            }
+        }
+        return isStarred;
+    }
+
+    @Override
+    public String toString() {
+        return this.starsUri.toString();
+    }
+
+    /**
+     * Repo owner holder.
+     */
+    private static class RepoOwnerHolder {
```

Comment:
@criske please remove this class, it's just a utility. Utility classes, helpers, holders, commons etc are what's wrong with today's OOP :) 

I prefer a little duplicated code (that split[] stuff) or maybe a small private utility method.

Reply:
> @criske please remove this class, it's just a utility. Utility classes, helpers, holders, commons etc are what's wrong with today's OOP :)

Care to explain why is bad, what's wrong with `java.util.Collections/Arrays` for example?

Anyway I think breaking that class into two member variables initialized at `GithubStars` constructor level would solve the duplicate code problem.

Reply:
Nothings wrong with collections/arrays. 

But this class doesn't implement anything, it's just a utility. 

A class should always implement something, an interface. In fact, the word ``class`` is wrong and misleading. It should be named ``implementation``